### ISSUE TITLE
Remove “{version} available” box

### ISSUE THREAD

**Issue by ausi** [contao/contao-manager#764]
Many of our customers struggle with the green “{version} available” box. It seems to many as something they have to fix. This causes problems in cases where they cannot get rid of this green box.

Would it make sense to remove this box completely?

An alternative could be something like “Check for Updates” that executes a dry-run and shows a list of all packages that are not up to date (would get updated).

**Comment by fritzmg** [comment]
> Would it make sense to remove this box completely?

It was a popular feature request though.

> An alternative could be something like “Check for Updates” that executes a dry-run and shows a list of all packages that are not up to date (would get updated).

Or just `composer outdated --direct`.

**Comment by ausi** [comment]
> > Would it make sense to remove this box completely?
> 
> It was a popular feature request though.

But what is this feature actually? 
If there is a green box, that does NOT mean that I can actually update. 
And if there is no green box, it does NOT mean that my installation is up to date.

So IMO this feature is not what most people think it is, and therefore causes confusion and a wrong feeling of “my installation is up to date” in most cases.

> though that will also tell you to update Contao 4.13 to 5.1 for example

The _new version outside your version constraint_ case is the only thing that works correctly at the moment, this could be kept as is IMO.

**Comment by aschempp** [comment]
> If there is a green box, that does NOT mean that I can actually update.
> And if there is no green box, it does NOT mean that my installation is up to date.

why not? I would think in 99% it is?

**Comment by ausi** [comment]
> I would think in 99% it is?

1. Every Contao 4.9 installation that has a `madeyourday/*` extension installed shows these green boxes even though the latest compatible version is already installed.

2. And I’d assume that most installations that show no green boxes are still out of date because some dependency has a new version since the last update.

So I’d say the percentage of this feature working as users expect it is maybe 90% for the first case and probably 1% for the second case.

**Comment by fritzmg** [comment]
Imho the second case - i.e. indirect dependencies - are a separate issue.

**Comment by mlwebworker** [comment]
Ich finde das grüne Kästchen als Hinweis, das es eine neuere Version außerhalb der eingetragenen Constraint-Einschränkungen gibt ganz praktisch. Ich glaube so etwas würde damals auch als Feature gewünscht.
Ich habe aber im Forum auch schon festgestellt, dass sich die Leute schwer tun mit dem Verständnis.

**Comment by ausi** [comment]
> Ich finde das grüne Kästchen als Hinweis, das es eine neuere Version außerhalb der eingetragenen Constraint-Einschränkungen gibt ganz praktisch.

Ja, dieses Feature funktioniert meiner Meinung nach einwandfrei und würde ich auch beibehalten.
Aber die grünen Boxen mit “1.2.3 verfügbar” (also innerhalb des eingetragenen Constraints) erzeugen sehr viel Verwirrung.

> Imho the second case - i.e. indirect dependencies being out of date - is a separate issue.

Then maybe my perception is wrong, or my customer base is different.
But in my view most of our customers interpret “no green box” as “the installation including all packages is up to date”.

**Comment by contaoacademy** [comment]
> As discussed in the Contao Show, it would be helpful, if the latest available versions of the installed packages and extensions would be shown in the overview and not only in the package detail view.
> 
> A problem is, that the latest version maybe couldn't be installed because the actual configuration doesn't allow this.
> 
> Example:
> 
> Contao 4.4 and PHP 7.1
> Contao 4.9 and PHP 7.4
> If a package needs Contao 4.9 or PHP 7.2 as minimum requirement, it isn't allowed to install this package for configuration 1. But the user always see, that there is a newer version. This could be confusing.
> 
> It should be discussed, if the benefit to show the latest version in overview is higher than confusing the user maybe not being able to install the latest version.

See: https://github.com/contao/contao-manager/issues/602

**Comment by aschempp** [comment]
we could likely fix 80% of the cases if we would check the Contao version compatibility as well as the constraint, right?

**Comment by ausi** [comment]
> we could likely fix 80% of the cases if we would check the Contao version compatibility as well as the constraint, right?

I guess so, probably even more than 90% of these cases are due to an incompatibility with the Contao version.

But in my opinion this is still a non-feature, and should be replaced with a “Check for Updates” button.

**Comment by fritzmg** [comment]
But I don't think that is what users actually want? I would assume they want to see at a glance whether there are updates available for all extensions.

**Comment by ausi** [comment]
> I would assume they want to see at a glance whether there are updates available for all extensions.

That might be, but it has to include the dependencies of the extensions (which it doesn’t) to make this useful. Otherwise metapackages would always seem to be up-to-date, even if all der dependent extensions are not up-to-date.

**Comment by ausi** [comment]
Another case where the green box causes more confusion than it helps IMO: https://community.contao.org/de/showthread.php?84936

**Comment by fritzmg** [comment]
But in this case it's just because the data is outdated. I assume this can be easily fixed by not showing it if the currently determined "newer" version is actually older than the currently installed version.
